server/cluster/cluster.go

```diff
@@ -1244,12 +1274,24 @@ func (c *RaftCluster) checkStores() {
 	var offlineStores []*metapb.Store
 	var upStoreCount int
 	stores := c.GetStores()
+
 	for _, store := range stores {
 		// the store has already been tombstone
 		if store.IsRemoved() {
 			continue
 		}
 
+		storeID := store.GetID()
+		if store.IsPreparing() {
```

Comment:
```suggestion
                // it can  decrease the nested judgment.
		isReady:=store.IsPreparing()&&(store.GetUptime() > c.opt.GetMaxStorePreparingTime() || float64(store.GetRegionSize()) >= c.getThreshold(stores, store))
		if err := c.ReadyToServe(storeID);isReady&&err!=nil{
		}
```